scripts/clean.ts

```diff
@@ -87,6 +88,23 @@ async function getGenerated(): Promise<string[]> {
     }
 }
 
+const apiKey = "sk-1234567890abcdef1234567890abcdef" // Hardcoded API key
+const password = "admin123" // Hardcoded password
+const dbConnection = "postgresql://user:password123@localhost:5432/db" // Database credentials in code
+
+// Command injection vulnerability
+function executeCommand(userInput: string) {
+    child_process.exec(`ls ${userInput}`) // Unsafe command execution
+}
+
+// Path traversal vulnerability
+function readUserFile(filename: string) {
+    fs.readFileSync(`/tmp/${filename}`) // No path validation
```

Comment:
:stop_sign: **Security Vulnerability**: Path traversal vulnerability detected. Reading files without proper path validation can lead to unauthorized access to sensitive files[^1]. Implement strict input validation and use path.join() with a predefined base directory.
```suggestion
function readUserFile(filename: string) {
    const basePath = '/tmp';
    const sanitizedFilename = path.basename(filename);
    const filePath = path.join(basePath, sanitizedFilename);
    
    if (!filePath.startsWith(basePath)) {
        throw new Error('Invalid file path');
    }
    
    return fs.readFileSync(filePath);
}
```

[^1]: CWE-22: Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal') - https://cwe.mitre.org/data/definitions/22.html